ros/src/sensing/filters/packages/points_preprocessor/nodes/points_concat_filter/points_concat_filter.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * fusion_filter.cpp
+ *
+ * Created on   : September 4, 2017
+ * Author       : Akihito OHSATO
+ */
+
+#include <ros/ros.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <message_filters/subscriber.h>
+#include <message_filters/synchronizer.h>
+#include <message_filters/sync_policies/approximate_time.h>
+#include <pcl_ros/point_cloud.h>
+#include <pcl_ros/transforms.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <pcl/point_types.h>
+#include <velodyne_pointcloud/point_types.h>
+#include <tf/tf.h>
+#include <tf/transform_listener.h>
+
+class PointsConcatFilter
+{
+public:
+  PointsConcatFilter();
+
+private:
+  typedef pcl::PointXYZI PointT;
+  typedef pcl::PointCloud<PointT> PointCloudT;
+  typedef sensor_msgs::PointCloud2 PointCloudMsgT;
+  typedef message_filters::sync_policies::ApproximateTime<PointCloudMsgT, PointCloudMsgT> SyncPolicy;
+  ros::NodeHandle nh_, pnh_;
+  message_filters::Subscriber<PointCloudMsgT> *sub1_, *sub2_;
+  message_filters::Synchronizer<SyncPolicy>* sync_;
+  ros::Publisher pub_;
+  tf::TransformListener tfl;
+  std::string sensor_frame;
+  void callback(const PointCloudMsgT::ConstPtr& msg1, const PointCloudMsgT::ConstPtr& msg2);
+};
+
+PointsConcatFilter::PointsConcatFilter() : nh_(), pnh_("~"), tfl(), sensor_frame("lidar_base")
+{
+  pnh_.param("sensor_frame", sensor_frame, sensor_frame);
+  sub1_ = new message_filters::Subscriber<PointCloudMsgT>(nh_, "/lidar0/points_raw", 1);
+  sub2_ = new message_filters::Subscriber<PointCloudMsgT>(nh_, "/lidar1/points_raw", 1);
+  sync_ = new message_filters::Synchronizer<SyncPolicy>(SyncPolicy(10), *sub1_, *sub2_);
```

Comment:
`ApproximateTime` will lead to a better frame rate, most of the scans are being dropped.
http://wiki.ros.org/message_filters#ApproximateTime_Policy